packages/strapi-admin/services/permission/engine.js

```diff
@@ -179,48 +181,91 @@ module.exports = conditionProvider => {
       // 1. Replace each condition name by its associated value
       const resolveConditions = map(conditionProvider.get);
 
-      // 2. Only keep the handler of each condition
-      const pickHandlers = map(prop('handler'));
-
-      // 3. Filter conditions, only keep objects and functions
-      const filterValidConditions = filter(isObject);
+      // 2. Filter conditions, only keep objects and functions
+      const filterValidConditions = filter(condition => isObject(condition.handler));
```

Comment:
can you rather check isFunction as you are looking at the handle now.

Reply:
A condition can be either  an object or a function (returning a boolean || an object) here

Reply:
As we now only accept functions, I've updated it.